src/test/java/seedu/address/logic/commands/GroupCommandTest.java

```diff
@@ -24,62 +31,107 @@
 import seedu.address.model.person.NusId;
 import seedu.address.model.person.Person;
 import seedu.address.testutil.GroupPersonDescriptorBuilder;
-import seedu.address.testutil.PersonBuilder;
-
-
 
 class GroupCommandTest {
 
     private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
 
     @Test
-    public void execute_allFieldsSpecifiedUnfilteredList_success() {
-        Person groupedPerson = model.getFilteredPersonList().get(0);
-        PersonBuilder personInList = new PersonBuilder(groupedPerson);
-        GroupCommand.GroupPersonDescriptor descriptor = new GroupPersonDescriptorBuilder(groupedPerson)
-                .build();
-        GroupCommand groupCommand = new GroupCommand(groupedPerson.getNusId(), descriptor);
+    public void execute_allFieldsSpecifiedUnfilteredListOnePerson_success() {
+        List<Person> groupedPerson = new ArrayList<>();
+        Set<NusId> nusIds = new HashSet<>();
+        groupedPerson.add(model.getFilteredPersonList().get(0));
+
+        for (Person person : groupedPerson) {
+            nusIds.add(person.getNusId());
+        }
+
+        GroupCommand.GroupPersonDescriptor descriptor = new GroupPersonDescriptorBuilder(groupedPerson).build();
+        GroupCommand groupCommand = new GroupCommand(nusIds, descriptor);
 
         String expectedMessage = String.format(GroupCommand.MESSAGE_GROUP_PERSON_SUCCESS,
                 Messages.format(groupedPerson));
 
         Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
-        expectedModel.setPerson(model.getFilteredPersonList().get(0), groupedPerson);
+        expectedModel.setPerson(model.getFilteredPersonList().get(0), groupedPerson.get(0));
+
+        assertCommandSuccess(groupCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Disabled
+    @Test
+    public void execute_allFieldsSpecifiedUnfilteredListMultiplePerson_success() {
+        List<Person> groupedPersons = new ArrayList<>();
+        Set<NusId> nusIds = new HashSet<>();
+        groupedPersons.add(model.getFilteredPersonList().get(0));
+        groupedPersons.add(model.getFilteredPersonList().get(1));
+        for (Person person : groupedPersons) {
+            nusIds.add(person.getNusId());
+        }
+
+        GroupCommand.GroupPersonDescriptor descriptor = new GroupPersonDescriptorBuilder(groupedPersons).build();
```

Comment:
I think you can import the nested class into the test class instead of using the dot operator